components/data-table/private/head.jsx

```diff
@@ -13,295 +13,304 @@ import HeaderCell from './header-cell';
 import InteractiveElement from '../interactive-element';
 import CellContext from '../private/cell-context';
 import TableContext from '../private/table-context';
-import contextHelper from './context-helper';
+import useContextHelper from './context-helper';
 
 // ## Constants
 import { DATA_TABLE_HEAD } from '../../../utilities/constants';
 
 const InteractiveCheckbox = InteractiveElement(Checkbox);
 
-/**
- * Used internally, provides header row rendering to the DataTable.
- */
-class DataTableHead extends React.Component {
-	// ### Display Name
-	// Always use the canonical component name as the React display name.
-	static displayName = DATA_TABLE_HEAD;
+// ### Prop Types
+const propTypes = {
```

Comment:
Most of the changes in this file are to refactor the component to a function so that it can use the context helper hook.